source/collimation.f90

```diff
@@ -2180,94 +2122,132 @@ subroutine collimate_start
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Created: 2019-09-11
+!  Updated: 2019-09-11
+!  Wrapper routine for main collimation call in tracking.
+! ================================================================================================ !
+subroutine collimate_trackThin(stracki, isColl)
 
-  use crcoall
-  use parpro
+  use mod_time
   use mod_common
   use mod_common_main
-  use mod_commons
   use mod_common_track
-  use mod_common_da
-  use numerical_constants, only : c5m4
-  use coll_db
 
-  implicit none
-
-  integer :: j
   real(kind=fPrec), intent(in) :: stracki
+  logical,          intent(in) :: isColl
 
-  if(cdb_elemMap(myix) > 0) then
-    nsig = cdb_cNSig(cdb_elemMap(myix))
-  else
-    nsig = cdb_defColGap
-  end if
+  integer j
 
-!++  Write trajectory for any selected particle
-  c_length = zero
+  if(isColl) then
 
-! SR, 23-11-2005: To avoid binary entries in 'amplitude.dat'
-  if( firstrun ) then
-    if(rselect.gt.0 .and. rselect.lt.65) then
-      do j = 1, napx
-        xj  = (xv1(j)-torbx(ie))/c1e3
-        xpj = (yv1(j)-torbxp(ie))/c1e3
-        yj  = (xv2(j)-torby(ie))/c1e3
-        ypj = (yv2(j)-torbyp(ie))/c1e3
-        pj  = ejv(j)/c1e3
+    call time_startClock(time_clockCOLL)
+    call collimate_start_collimator(stracki)
+    call collimate_do_collimator(stracki)
+    call collimate_end_collimator(stracki)
+    call time_stopClock(time_clockCOLL)
 
-        if(iturn.eq.1.and.j.eq.1) then
-          sum_ax(ie)=zero
-          sum_ay(ie)=zero
-        end if
+  else
 
-!-- DRIFT PART
-        if(stracki.eq.0.) then
-          if(iexact) then
-            zpj = sqrt(one-xpj**2-ypj**2)
-            xj  = xj + half*c_length*(xpj/zpj)
-            yj  = yj + half*c_length*(ypj/zpj)
-          else
-            xj  = xj + half*c_length*xpj
-            yj  = yj + half*c_length*ypj
-          end if
-        end if
+    gammax = (one + talphax(ie)**2)/tbetax(ie)
+    gammay = (one + talphay(ie)**2)/tbetay(ie)
 
-        gammax = (one + talphax(ie)**2)/tbetax(ie)
-        gammay = (one + talphay(ie)**2)/tbetay(ie)
+    do j=1,napx
+      xj  = (xv1(j)-torbx(ie))/c1e3
+      xpj = (yv1(j)-torbxp(ie))/c1e3
+      yj  = (xv2(j)-torby(ie))/c1e3
+      ypj = (yv2(j)-torbyp(ie))/c1e3
+      pj  = ejv(j)/c1e3
 
-        if(part_abs_pos(j).eq.0 .and. part_abs_turn(j).eq.0) then
-          nspx = sqrt(abs( gammax*(xj)**2 + two*talphax(ie)*xj*xpj +tbetax(ie)*xpj**2 )/myemitx0_collgap)
-          nspy = sqrt(abs( gammay*(yj)**2 + two*talphay(ie)*yj*ypj +tbetay(ie)*ypj**2 )/myemity0_collgap)
-          sum_ax(ie)   = sum_ax(ie) + nspx
-          sqsum_ax(ie) = sqsum_ax(ie) + nspx**2
-          sum_ay(ie)   = sum_ay(ie) + nspy
-          sqsum_ay(ie) = sqsum_ay(ie) + nspy**2
-          nampl(ie)    = nampl(ie) + 1
-        else
-          nspx = zero
-          nspy = zero
+      if(firstrun) then
+        if(iturn == 1 .and. j == 1) then
+          sum_ax(ie) = zero
+          sum_ay(ie) = zero
         end if
+      end if
```

Comment:
shouldn't we bring this loop out of the `do j=1,napx` one? still checking `if(firstrun)` and `if (iturn==1)`

Reply:
Fixed